service/client.ts

```diff
@@ -23,6 +24,42 @@ import isGrpcError from "utils/isGrpcError";
 
 const URL = process.env.NEXT_PUBLIC_API_BASE_URL;
 
+async function http<T>(endpoint: string, configOverride: RequestInit): Promise<T> {
+  const url = path.join(URL, endpoint)
+  const config = {
+    headers: {
+      "content-type": "application/json"
+    },
+    ...configOverride
+  }
+  const request = new Request(url, config)
+  const response = await fetch(request)
+
+  // if(!response.ok) {
+  //   throw new Error(response.statusText)
+  // }
+
+  // may error if there is no body, return empty array
+  return response.json().catch(() => ({}))
+}
+
+export async function get<T>(path: string, config?: RequestInit): Promise<T> {
+  const init = {method: 'get', ...config}
+  return await http<T>(path, init)
+}
+
+export async function post<T, U>(path: string, body: T, config?: RequestInit): Promise<U> {
```

Comment:
```suggestion
export async function post<T, U>(path: string, body: T, config?: Omit<RequestInit, 'body'>): Promise<U> {
```
To prevent accidentally overriding `body` if you pass it through the `config` object later. Same comment for the `put` request helper